lib/file.go

```diff
@@ -24,7 +25,21 @@ func FileCopy(dest string, source string) error {
 			return errors.Wrapf(err, "Coudn't read link %s", source)
 		}
 
-		return os.Symlink(target, dest)
+		if err = os.Symlink(target, dest); err != nil {
+			return errors.Wrapf(err, "Couldn't symlink %s->%s", source, target)
+		}
+
+		if err = os.Lchown(dest, uid, gid); err != nil {
+			return errors.Wrapf(err, "Couldn't set symlink ownership %s", dest)
+		}
+
+		if mode != nil {
+			if err = os.Chmod(dest, *mode); err != nil {
```

Comment:
I think this is very wrong.  chmod on a symbolic link, which you've just created dest as, does not exist.  Instead, golang's os.Chmod does (per https://pkg.go.dev/os#Chmod) :

> Chown changes the numeric uid and gid of the named file. If the file is a symbolic link, it changes the uid and gid of the link's target

Reply:
Note you should indeed Lchown it, as you are, so that part is good.

Reply:
```
$ ln -sfv non-existent bad-link
'bad-link' -> 'non-existent'

$ ls -al bad-link
lrwxrwxrwx 1 rchincha ifc 12 Dec 14 20:33 bad-link -> non-existent

$ chmod 755 bad-link
chmod: cannot operate on dangling symlink 'bad-link'
```